manifests/sar_metric.pp

```diff
@@ -1,33 +1,34 @@
+# Collect System Metrics
 define puppet_metrics_collector::sar_metric (
-  String                    $output_dir,
-  String                    $scripts_dir,
-  Enum['absent', 'present'] $metric_ensure = 'present',
-  String                    $metrics_type = $title,
-  String                    $cron_minute = '*/5',
-  Integer                   $retention_days = 90,
+  String                    $metrics_type              = $title,
+  Enum['absent', 'present'] $metric_ensure             = 'present',
+  String                    $cron_minute               = '*/5',
+  Integer                   $retention_days            = 90,
+  Integer                   $collection_frequency      = 5, # minutes
   Integer                   $polling_frequency_seconds = 1,
-  Integer                   $collection_frequency = 5, #minutes
-  String                    $metric_script_file = 'generate_system_metrics',
+  String                    $metric_script_file        = 'system_metrics',
 ) {
 
-  $metrics_output_dir = "${output_dir}/${metrics_type}"
+  $metrics_output_dir = "${puppet_metrics_collector::system::output_dir}/${metrics_type}"
 
-  $_metric_ensure = $metric_ensure ? {
+  $metrics_output_dir_ensure = $metric_ensure ? {
     'present' => directory,
     'absent'  => absent,
   }
 
   file { $metrics_output_dir :
-    ensure => $_metric_ensure,
+    ensure => $metrics_output_dir_ensure,
   }
 
-  $script_file_name = "${scripts_dir}/${metric_script_file}"
+  $metric_script_file_path = "${puppet_metrics_collector::system::scripts_dir}/${metric_script_file}"
   $file_interval_seconds = $collection_frequency * 60
 
-  $metrics_command = join(["${script_file_name} --metric_type ${metrics_type}",
+  $metrics_command = join([$metric_script_file_path,
+                            " --metric_type ${metrics_type}",
                             " --file_interval ${file_interval_seconds}",
                             " --polling_interval ${polling_frequency_seconds}",
-                            " --metrics_dir ${output_dir}"], '')
+                            " --metrics_dir ${puppet_metrics_collector::system::output_dir}"
+                            ], '')
```

Comment:
I do not know! Yes?

Reply:
All output `> /dev/null 2>&1` or just STDERR `2>/dev/null`. ?

Reply:
Hmm. Maybe just STDOUT? That way STDERR could go to mail?

Reply:
I'm not familiar with that code, so don't know what would be appropriate.

Reply:
Good point. Let's just dump it all to /dev/null so it does not spam the user.

Reply:
Well, I do se `pe_metric.pp` just does `> /dev/null`
Shall we model that, so errors don't disappear?

Reply:
```
      $metrics_command = $server_type ? {
        'influxdb'   => "${full_metrics_command} --influx-db ${server_db} > /dev/null",
        'graphite'   => "${full_metrics_command} > /dev/null",
        'splunk_hec' => "${base_metrics_command} | /opt/puppetlabs/bin/puppet splunk_hec --sourcetype puppet:metrics --pe_metrics > /dev/null",
        default      => "${full_metrics_command} > /dev/null",
      }
```

Reply:
That aligns with STDOUT to `/dev/null` and STDERR to mail (or however they have configured it). It sounds like the best plan since it should allow for the user to know about errors and correct them without being spammed with STDOUT messages.

Reply:
Done ... without force!